### PR TITLE
Fix CI errors

### PR DESCRIPTION
**Pull request by tamiroh**
To fix the CI errors, I mainly updated the svelte config in `.eslintrc`.

See comments for other fixes and additional information.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -5,28 +5,37 @@
...
             "files": [
                 "*.svelte"
             ],
-            "processor": "svelte3/svelte3"
+            "parser": "svelte-eslint-parser",
+            "parserOptions": {
+                "parser": "@typescript-eslint/parser"
+            },
+            "rules": {
+                "no-undef": "off" // TODO: remove this line
```
Comment (tamiroh) [inline .eslintrc:20]: Here I temporarily suppressing the `no-undef` errors, which currently occurs for global objects such as `window`, `app` and so on. I can't solve this problem right now, so this is a temporary work around.

File: src/ui/sourceControl/components/stagedFileComponent.svelte [hunk 2]
```diff
@@ -11,7 +11,6 @@
     export let view: GitView;
     export let manager: GitManager;
     let buttons: HTMLElement[] = [];
-    $: formattedPath = change.vault_path;
```
Comment (tamiroh) [inline src/ui/sourceControl/components/stagedFileComponent.svelte:14]: I removed this line to suppress the following error.

```
14:8   error  'formattedPath' is defined but never used  @typescript-eslint/no-unused-vars
```

This variable is not actually referenced and should not affect the behavior.

File: src/main.ts [hunk 3]
```diff
@@ -1251,7 +1251,7 @@ export default class ObsidianGit extends Plugin {
             this.app.workspace.iterateAllLeaves((leaf) => {
                 if (
                     leaf.view instanceof MarkdownView &&
-                    leaf.view.file.path == file.path
+                    (leaf.view.file as any).path == file.path
```
Comment (Vinzent03) [inline src/main.ts:1254]:
```suggestion
                    leaf.view.file?.path == file.path
```

**Review by Vinzent03 — CHANGES_REQUESTED** [review]

**Comment by Vinzent03** [comment]
Thanks for the fixes!

**Pull request merged**